solr/core/src/java/org/apache/solr/cloud/api/collections/CollectionHandlingUtils.java

```diff
@@ -506,30 +507,23 @@ private static NamedList<Object> waitForCoreAdminAsyncCallToComplete(
           }
 
           String r = (String) srsp.getSolrResponse().getResponse().get("STATUS");
-          if (r.equals("running")) {
```

Comment:
Why don't we change this to a `switch` statement?  Even IntelliJ is recommending this :-)  It seems an underlying bug here would have been averted had a switch statement been used as it sort of forces you to consider all the enum values.

Reply:
We don't check an enum, but string values.
I often feel a `switch` statement on strings is error prone, and doesn't protect from regressions.
What do you think?

Reply:
I may miss something. Do you suggest to introduce a new enum for this? I don't think there is one for request statuses.
Here `PENDING` is just a constant string.

Reply:
`org.apache.solr.client.solrj.response.RequestStatusState` but it looks like I confused collection async status with core async status.  Too bad they don't use the same enum!